public/app/features/logs/components/LogRowContextProvider.tsx

```diff
@@ -93,8 +93,14 @@ export const getRowContexts = async (
           // ns which came before but they come in the response that search for logs after. This means right now
           // we will show those as if they came after. This is not strictly correct but seems better than losing them
           // and making this correct would mean quite a bit of complexity to shuffle things around and messing up
-          //counts.
-          if (idField.values.get(fieldIndex) === row.uid) {
+          // counts.
+          // We add the refId of the original dataFrame, since uids contain those now and the uids in the context-dataframe
+          // do not have a refid.
+          // To be compatible to other datasources we check for both IDs with and without refId.
+          if (
+            idField.values.get(fieldIndex) === row.uid ||
+            `${idField.values.get(fieldIndex)}${row.dataFrame.refId}` === row.uid
+          ) {
```

Comment:
@gabor I am unsure about this. Should we check for both or just 
```
`${idField.values.get(fieldIndex)}${row.dataFrame.refId}` === row.uid
```

Reply:
Nevermind. I made an other approach by adding the original refid to the context-query.

Reply:
https://github.com/grafana/grafana/pull/56370/files#diff-2e4d3a3dbd431f921794de33720c6fc3ac7a8a2f6dbde31424e487080bb572c1R593